lib/kafkaClient.js

```diff
@@ -413,6 +413,81 @@ KafkaClient.prototype.getApiVersions = function (broker, cb) {
   broker.write(request);
 };
 
+KafkaClient.prototype.getListGroups = function (callback) {
+  if (!this.ready) {
+    return callback(new Error('Client is not ready (getListGroups)'));
+  }
+  const brokers = this.brokerMetadata;
+  async.mapValuesSeries(brokers,
+    (brokerMetadata, brokerId, cb) => {
+      const broker = this.brokerForLeader(brokerId);
+      if (!broker || !broker.isConnected()) {
+        return callback(new errors.BrokerNotAvailableError('Broker not available (getListGroups)'));
+      }
+
+      const correlationId = this.nextId();
+      const request = protocol.encodeListGroups(this.clientId, correlationId);
+
+      this.queueCallback(broker.socket, correlationId, [
+        protocol.decodeListGroups,
+        cb,
+        this.options.versions.requestTimeout
+      ]);
+      broker.write(request);
+    }, (err, results) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+      // removing {error: null} before merging
+      results = _.map(results, result => _.omitBy(result, _.isNull));
+
+      callback(null, _.merge({}, ...(_.values(results))));
```

Comment:
We are still supporting node 4 so you can't use spread operators here.

Reply:
I'll change it to use `.apply` instead